interfaces/policy/basedeclaration_test.go

```diff
@@ -1043,6 +1047,10 @@ func (s *baseDeclSuite) TestPlugInstallation(c *C) {
 				comm := Commentf("%s by %s snap", iface.Name(), name)
 				if ok {
 					c.Check(err, IsNil, comm)
+					// only restricted plugs can use the AppArmor
+					// unconfined profile mode so check that this
+					// plug is not using it
+					c.Assert(interfaces.StaticInfoOf(iface).AppArmorUnconfinedPlugs, Equals, false)
```

Comment:
Updated in fdbda18b31 but I left the one which checks the plug side against the known list of `restricted` interfaces since this makes more sense to do on a case-by-case basis IMO.